src/mhd2xds.coffee

```diff
@@ -22,6 +39,13 @@ exports.mhd1Metadata2Xds = (metadata) ->
     seconds = '0' + seconds
   now = "#{year}#{month}#{day}#{hours}#{minutes}#{seconds}"
 
+  if metadata.documentEntry.patientId.indexOf '&' is -1 and metadata.documentEntry.patientId.indexOf 'ZAF' isnt -1
+    sourceId = metadata.documentEntry.patientId.replace 'ZAF', 'ZAF&0.0.0&ISO'
```

Comment:
On chatting on hipcaht we came up with the solution of encoding the as follow to be standards compiant: `#{patientId}^^^&#{patientIdType}-#{patientIdAssigningAuthority}-526ef9c3-6f18-420a-bc53-9b733920bc67&ISO`